### PR TITLE
FEATURE: add functionality to clean up form data

### PR DESCRIPTION
**Pull request by andrehoffmann30**

### CODE REVIEW

File: Classes/Domain/Repository/FormDataRepository.php [hunk 1]
```diff
@@ -163,4 +163,15 @@ protected function findAllUniqueSiteNames(): array
 
         return $sites;
     }
+
+    /**
+     * @throws InvalidQueryException
+     */
+    public function findAllOlderThan(\DateTime $date): QueryResultInterface
```
Comment (daniellienert) [inline Classes/Domain/Repository/FormDataRepository.php:170]: Why not using a delete query directly here?

File: Classes/Domain/FormDataCleanup/FormDataCleanupService.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+use Doctrine\ORM\ORMException;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Persistence\Exception\IllegalObjectTypeException;
+use Neos\Flow\Persistence\Exception\InvalidQueryException;
+use PunktDe\Form\Persistence\Domain\Model\FormData;
+use PunktDe\Form\Persistence\Domain\Repository\FormDataRepository;
+
+class FormDataCleanupService
```
Comment (daniellienert) [inline Classes/Domain/FormDataCleanup/FormDataCleanupService.php:18]: Could possibly be removed if a delete query is used.

File: Configuration/Settings.yaml [hunk 3]
```diff
@@ -34,3 +34,6 @@ PunktDe:
         senderName: Scheduled Form Data Export
         senderMailAddress: noreply@localhost
         subject: "Latest data of form {formIdentifier}"
+
+      deleteFormData:
```
Comment (daniellienert) [inline Configuration/Settings.yaml:38]: "formDataCleanup" like in the service sounds better

File: Configuration/Settings.yaml [hunk 4]
```diff
@@ -34,3 +34,6 @@ PunktDe:
...
+        dateInterval: 'P30D'
```
Comment (daniellienert) [inline Configuration/Settings.yaml:39]: "dateInterval"  would better be "retentionPeriod"

**Comment by daniellienert** [comment]
The feature needs some reference in the readme

File: Classes/Domain/FormDataCleanup/FormDataCleanupService.php (hunk 2 above)
Comment (andrehoffmann30) [reply Classes/Domain/FormDataCleanup/FormDataCleanupService.php:18]: I would like to leave this service. I think with this service we provide an easier way to integrate this functionality into other code.

**Comment by andrehoffmann30** [comment]
I added a reference for this feature to the readme.

**Review by daniellienert — APPROVED** [review]

**Pull request merged**